server/src/test/java/org/elasticsearch/action/admin/cluster/coordination/MasterHistoryActionTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.coordination;
+
+import org.elasticsearch.Version;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.cluster.coordination.MasterHistory;
+import org.elasticsearch.cluster.coordination.MasterHistoryService;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.EqualsHashCodeTestUtils;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class MasterHistoryActionTests extends ESTestCase {
+    public void testSerialization() {
+        List<DiscoveryNode> masterHistory = List.of(
+            new DiscoveryNode("_id1", buildNewFakeTransportAddress(), Version.CURRENT),
+            new DiscoveryNode("_id2", buildNewFakeTransportAddress(), Version.CURRENT)
+        );
+        MasterHistoryAction.Response response = new MasterHistoryAction.Response(masterHistory);
+        EqualsHashCodeTestUtils.checkEqualsAndHashCode(
```

Comment:
Nit: this 2-argument version of `checkEqualsAndHashCode` doesn't detect bugs where the `.equals()` method isn't discriminating enough (e.g. it just checks the types are compatible). Probably not a big deal here, but in general we should use the 3-argument version with a mutation function.